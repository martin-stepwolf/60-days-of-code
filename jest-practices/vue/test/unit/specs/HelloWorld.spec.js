import Vue from 'vue';
import HelloWorld from '@/components/HelloWorld';

describe('Test in Vue', () => {
  test('should render correct contents', () => {
    const Constructor = Vue.extend(HelloWorld)
    const vm = new Constructor().$mount()
    expect(vm.$el.querySelector('.hello h1').textContent)
      .toEqual('Hello Vue :)')
  })

  test('Create a snapshot', () => {
    const Constructor = Vue.extend(HelloWorld)
    const vm = new Constructor().$mount()
    expect(vm.$el).toMatchSnapshot()
  })

})