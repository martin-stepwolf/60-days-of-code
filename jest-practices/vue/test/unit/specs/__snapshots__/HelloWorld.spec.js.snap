// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AÃ±adiendo pruebas a Vue Creamos un snapshot en Vue 1`] = `
<div
  class="hello"
>
  <h1>
    Hello Vue :)
  </h1>
   
  <ul>
    <li>
      <a
        href="#"
        target="_blank"
      >
        
        Esto es un Link :) 
      
      </a>
    </li>
  </ul>
</div>
`;

exports[`HelloWorld.vue has the expected html structure 1`] = `
<div
  class="hello"
>
  <h1>
    Hello Vue
  </h1>
   
  <ul>
    <li>
      <a
        href="#"
        target="_blank"
      >
        
        Esto es un Link :) 
      
      </a>
    </li>
  </ul>
</div>
`;
